frb_codegen/src/ir/ty_unencodable.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{ir::*, target::Target};
+use quote::ToTokens;
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum ArgsRefs<'a> {
+    Generic(&'a [IrType]),
+    Signature(&'a [IrType]),
+}
+
+pub trait ArgsUnpackable {
+    fn unpack(&self) -> Vec<(&str, Option<ArgsRefs>)>;
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct IrTypeUnencodable {
+    pub underlying_type: Box<syn::Type>,
+    pub segments: Vec<(String, Option<Args>)>,
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum Args {
+    Generic(Vec<IrType>),
+    Signature(Vec<IrType>),
+}
+
+impl ArgsUnpackable for Vec<(String, Option<Args>)> {
+    fn unpack(&self) -> Vec<(&str, Option<ArgsRefs>)> {
```

Comment:
uber-nit: looks like this wants to "take `&`"? then not sure why it is called "unpack"

Reply:
Fixed, I called it "splay."